engineapi/types.go

```diff
@@ -259,9 +271,9 @@ func GetEngineEndpoint(volume *Volume, ip string) (string, error) {
 	}
 
 	switch volume.Frontend {
-	case devtypes.FrontendTGTBlockDev:
+	case iscsidevtypes.FrontendTGTBlockDev, spdkdevtypes.FrontendSPDKTCPBlockdev:
 		return volume.Endpoint, nil
-	case devtypes.FrontendTGTISCSI:
+	case iscsidevtypes.FrontendTGTISCSI, spdkdevtypes.FrontendSPDKTCPNvmf:
 		if ip == "" {
```

Comment:
The message below should be updated for different stacks.

```go
return "", fmt.Errorf("iscsi endpoint %v is missing ip", volume.Endpoint)

return EndpointISCSIPrefix + ip + ":" + DefaultISCSIPort + "/" + volume.Endpoint + "/" + DefaultISCSILUN, nil

```

Reply:
Addressed. Thank you.